components/brave_ads/core/internal/ads_impl.cc

```diff
@@ -93,6 +94,45 @@ void AdsImpl::Shutdown(ShutdownCallback callback) {
   std::move(callback).Run(/*success=*/true);
 }
 
+void AdsImpl::GetInternals(GetInternalsCallback callback) {
+  if (task_queue_.should_queue()) {
+    return task_queue_.Add(base::BindOnce(&AdsImpl::GetInternals,
+                                          weak_factory_.GetWeakPtr(),
+                                          std::move(callback)));
+  }
+
+  // TODO(tmancey): Decouple.
+  database::table::CreativeSetConversions database_table;
+  database_table.GetActive(base::BindOnce(&AdsImpl::GetActiveCallback,
+                                          weak_factory_.GetWeakPtr(),
+                                          std::move(callback)));
+}
+
+// TODO(tmancey): Decouple.
```

Comment:
Would be great to remove the `TODO` and move this logic out of `AdsImpl`, thanks

Reply:
Filed an issue for it: https://github.com/brave/brave-browser/issues/42551